### DISCUSSION TITLE
Under what conditions is KAMAL_ROLE available to hooks?

### DISCUSSION THREAD

**Discussion by hcpss-banderson** [basecamp/kamal#1219] [Q&A]
I tried `kamal deploy --roles=aes` but the pre-deploy hook still does not have access to the KAMAL_ROLE variable.

For example:

```bash
#!/bin/sh
echo "$KAMAL_ROLE" > ./role.txt
````

This gives me an empty file.

**Comment by tuladhar** [comment]
Can you run `env` command in the script to dump environment variable set by Kamal.

```sh
#!/bin/sh
# dump env
env
echo "$KAMAL_ROLE" > ./role.txt
```

**Reply by hcpss-banderson** [reply]
Empty file.

**Reply by hcpss-banderson** [reply]
Sorry, I see it should be in the output.

All my secrets are there, but the only KAMAL_ variable i have is KAMAL_SERVICE.

**Reply by tuladhar** [reply]
Interesting, what version of Kamal are you running:
```
$ kamal version
```
